File: server/src/main/java/net/progruzovik/dissent/socket/Sender.java
package net.progruzovik.dissent.socket;

import net.progruzovik.dissent.model.message.Message;
import org.springframework.lang.NonNull;
import org.springframework.web.socket.WebSocketSession;

public interface Sender {

    void setSession(@NonNull WebSocketSession session);

    <T> void sendMessage(Message<T> message);
}

File: server/src/main/kotlin/net/progruzovik/dissent/socket/Sender.kt
package net.progruzovik.dissent.socket

import net.progruzovik.dissent.model.socket.ServerSubject
import org.springframework.web.reactive.socket.WebSocketSession

interface Sender {

    fun setUpSession(session: WebSocketSession)

    fun <T> sendMessage(subject: ServerSubject, data: T?)

    fun sendMessage(subject: ServerSubject) = sendMessage(subject, null)
}
